File: TrapprKeepr/Models/Vault.cs
namespace TrapprKeepr.Models;

public class Vault
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public bool? IsPrivate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string CreatorId { get; set; }
    public Profile Creator { get; set; }
}
public class VaultVkeepsViewModel : Vault
{
    public int VaultKeepId { get; set; }
    public string ProfileId { get; set; }
}

File: TrapprKeepr.client/src/models/Vault.js
export class Vault {
    constructor(data) {
        this.id = data.id
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.isPrivate = data.isPrivate || false
        this.creatorId = data.creatorId
        this.creator = data.creator
        this.createdAt = data.createdAt
        this.updatedAt = data.updatedAt
        // NOTE classed on the VaultKeeps
        this.isVault = 'this is a classed vault'

    }
}